src/gui/settingsdialog.cpp

```diff
@@ -158,12 +167,14 @@ SettingsDialog::SettingsDialog(ownCloudGui *gui, QWidget *parent)
 
     customizeStyle();
 
-    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint & Qt::Window);
```

Comment:
On Linux I was already carefully resizing the settings dialog to be quasi-full-screen so I could monitor the progress of longer synchronization actions. FWIW, even if you do eventually downsize the dialog, it would be helpful in the meantime to have this fix available, because it's unclear when that change would actually be implemented.

(I also do have *thoughts* on the design of the popover and the settings dialog, but I haven't found a good place to put them, yet.)

Reply:
> (I also do have _thoughts_ on the design of the popover and the settings dialog, but I haven't found a good place to put them, yet.)

Best option is open issues on the repository, make sure to tag them design and extensively poke @jancborchardt about them. :-)

Reply:
I just haven't gotten around to it, and I didn't know if there was an existing thread like there is with the first-run wizard.

Reply:
> this being a regular window instead of a dialog (with menubar and all)

Right, so what are the drawbacks of allowing maximizing? The default size would still be the same I assume. And resizing is still possible. Is having a menubar a must then or can it be disabled?

@elsiehupp screenshots of how it looks like with your change would be very appreciated. :) Especially also on Windows or Linux where the menu is part of the window unlike in macOS.

> (I also do have thoughts on the design of the popover and the settings dialog, but I haven't found a good place to put them, yet.)

Really good stuff @elsiehupp! :) It’s probably good to open new issues for that anyway since we didn’t talk about this in some time. New input would be great.